alvr/client_core/src/graphics/stream.rs

```diff
@@ -216,24 +224,51 @@ impl StreamRenderer {
             .device
             .create_command_encoder(&Default::default());
 
-        for (view_idx, swapchain_idx) in swapchain_indices.iter().enumerate() {
+        for (view_idx, view_params) in view_params.iter().enumerate() {
             let mut render_pass = encoder.begin_render_pass(&RenderPassDescriptor {
                 label: None,
                 color_attachments: &[Some(RenderPassColorAttachment {
-                    view: &self.views_objects[view_idx].render_target[*swapchain_idx as usize],
+                    view: &self.views_objects[view_idx].render_target
+                        [view_params.swapchain_index as usize],
                     resolve_target: None,
                     ops: wgpu::Operations {
-                        load: LoadOp::Clear(wgpu::Color::RED),
+                        load: LoadOp::Clear(Color::BLACK),
                         store: StoreOp::Store,
                     },
                 })],
                 ..Default::default()
             });
 
+            let fov = view_params.fov;
+
+            let tanl = f32::tan(fov.left);
+            let tanr = f32::tan(fov.right);
+            let tanu = f32::tan(fov.up);
+            let tand = f32::tan(fov.down);
+
+            let width = tanr - tanl;
+            let height = tanu - tand;
+
+            // The image is at z = -1.0, so we use tangents for the size
+            let model_mat =
+                Mat4::from_translation(Vec3::new(width / 2.0 + tanl, height / 2.0 + tand, -1.0))
+                    * Mat4::from_scale(Vec3::new(width, height, 1.));
+            let view_mat = Mat4::from_quat(view_params.reprojection_rotation).inverse();
+            let proj_mat = super::projection_from_fov(view_params.fov);
+
+            let transform = proj_mat * view_mat * model_mat;
```

Comment:
Are you sure this math is accurate enough to not cause aliasing when using the identity quat? Or does the identity quaternion have 0s in all the right places to make all this math be ignored? Because otherwise I think it'd be better to pass the projection in as an option and only do all of this when we actually want to do it.

Reply:
There would be some cancelling out when rendering the quad with that specific size and using that specific projection matrix. Both use the same tangent values.

Reply:
I can say that empirically using a quat identity the quality is the same as no reprojection. With any other quat i can see the image quality is degraded

Reply:
Well I don't have the time to verify, so you'll have to be right I guess.